Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/testresult/pending/SubmissionTestResultPendingFragment.kt

```diff
@@ -43,7 +45,8 @@ class SubmissionTestResultPendingFragment : Fragment(R.layout.fragment_submissio
         pendingViewModel.testState.observe2(this) { result ->
             result.deviceUiState.withFailure {
                 if (it is CwaWebException) {
-                    DialogHelper.showDialog(buildErrorDialog(it))
+                    dialog?.dismiss()
+                    dialog = DialogHelper.showDialog(buildErrorDialog(it))
```

Comment:
We observe a failure here twice and therefore there is flickering, because the dialog is quickly shown, dismissed and shown again. 

So I would suggest:

```suggestion
                    if (dialog == null || dialog?.isShowing == false) {
                        dialog = DialogHelper.showDialog(buildErrorDialog(it))
                    }
```

The better solution though would be if we wouldn't receive the failure here twice. But this would be needed to fix in another layer of the app I guess.

Reply:
There are different error dialogs, otherwise your suggestion would solve the issue. I will investigate why the state is issued twice.

Reply:
Are you sure there are different error dialogs? 

Since we have the `if (it is CwaWebException)` condition before showing a dialog here, `buildErrorDialog(it)` will always build the same dialog, or am I missing something?

Reply:
I also see the duplicate dialogs. I think it happens because the `SubmissionRepository` still holds the previous data to which we subscribe when opening the screen, but then opening the screen also causes a refresh too, via `refreshDeviceUIState`.

> Since we have the if (it is CwaWebException) condition before showing a dialog here, buildErrorDialog(it) will always build the same dialog, or am I missing something?

We can probably remove the `else {` branch, the only difference seems to be the web-exception status code and that is available in any case via the base exception class `CwaWebException`.

The `else` branch may be a left over from previous submission flow refactoring.

Reply:
I adjusted the logic so that there is only one error dialog, no old values trigger the dialog when opening the screen multiple times/ putting app to background

Reply:
The root cause it that the deviceUIStateFlow is a mixture of the result of a specific request (success or failure) and the current state of the corona test.